web/src/components/JobPane.vue

```diff
@@ -0,0 +1,219 @@
+<template>
+  <v-card class="pa-3">
+    <v-select
+      v-model="selectedDataset"
+      label="Select data for analysis"
+      :items="currentDatasets"
+      item-text="name"
+      clearable
+    />
+    <span v-if="selectedDataset && !jobSpawned">
+      Please note that custom channel coloration is not used in image analysis.
+    </span>
+    <div
+      v-if="selectedDataset && !jobSpawned"
+      style="display:flex; justify-content:space-between"
+    >
+      <v-switch
+        v-model="entireImage"
+        label="Analyze entire image"
+      />
+      <div
+        v-if="savedNotification && !entireImage"
+        style="display: inline"
+      >
+        {{ savedNotification }}
+      </div>
+      <v-btn
+        v-if="!selectionMode && !entireImage"
+        @click="() => store.commit.setSelectionMode(true)"
+      >
+        Select Region
+      </v-btn>
+      <v-btn
+        v-if="selection && !entireImage"
+        color="primary"
+        @click="useSelection"
+      >
+        Use Selection
+      </v-btn>
+      <v-btn
+        v-if="selectionMode && !entireImage"
+        color="secondary"
+        @click="() => store.commit.setSelectionMode(false)"
+      >
+        Cancel
+      </v-btn>
+    </div>
+    <div v-if="selectedDataset && !jobSpawned">
+      <v-select
+        v-model="selectedJobType"
+        label="Available Job Types"
+        :items="store.state.jobTypes"
+      >
+        <template v-slot:selection="{ item }">
+          {{ item.name.replace(/_/g, ' ') }}
+        </template>
+        <template v-slot:item="{ item, attrs, on }">
+          <v-list-item
+            v-bind="attrs"
+            v-on="on"
+          >
+            <v-list-item-content>
+              <v-list-item-title>
+                {{ item.name.replace(/_/g, ' ') }}
+                <div class="job-type-description">
+                  {{ item.description }}
+                </div>
+              </v-list-item-title>
+            </v-list-item-content>
+          </v-list-item>
+        </template>
+      </v-select>
+      <div v-if="selectedJobType">
+        <v-form v-model="inputFormValid">
+          <v-jsonschema-form
+            :schema="selectedJobType.schema"
+            :model="jobInputs"
+            :options="schemaOptions"
+          />
+        </v-form>
+        <v-btn
+          v-if="inputFormValid"
+          color="primary"
+          @click="spawnJob"
+        >
+          Spawn Job
+        </v-btn>
+      </div>
+    </div>
+    <div v-if="jobSpawned">
+      Job spawned successfully! Results will appear as a Pin when the job run is complete.
+    </div>
+  </v-card>
+</template>
+
+<style scoped>
+.job-type-description {
+  color: gray;
+}
+</style>
+
+<script>
```

Comment:
In other components we use TypeScript. @waxlamp Is this still something we want to continue requiring in Phase I?

Reply:
The conversion is simple enough, we may as well continue typescript. Changed in [d238ad3](https://github.com/atlascope/atlascope/pull/140/commits/d238ad3d22371cf03a2fe54af101f51cbeee748e)

Reply:
(Sticking with TypeScript is a good choice even for Phase I, because it requires a relatively small up-front investment of time to avoid relatively large time losses later.)